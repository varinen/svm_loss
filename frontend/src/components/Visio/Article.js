import React from "react";
import MathJax from 'react-mathjax';
import {Typography} from "@material-ui/core";

const formulas = {
    f1: `W_{0,0}x_0+W_{0,1}x_1+b_0`,
    f2: `(x_0,x_1)`,
    f3: `(W_{0,0},W_{0,1})`
};


export default () => (
    <MathJax.Provider>
        <article>
            <Typography variant="body2" color="textSecondary" component="p">
                Data points are shown as circles colored by their class (red/green/blue). The background
                regions
                are colored by whichever class is most likely at any point according to the current weights.
            </Typography>

            <Typography variant="body2" color="textSecondary" component="p">
                Each classifier is visualized by a line that indicates its zero score level set. For
                example,
                the blue classifier computes scores
                as <MathJax.Node inline
                                 formula={formulas.f1}/> and
                the blue line shows the set of
                points <MathJax.Node inline
                                     formula={formulas.f2}/> that
                give score of zero. The blue arrow draws
                the
                vector <MathJax.Node inline
                                     formula={formulas.f3}/> ,
                which shows the direction of score increase and its length is proportional to how steep the
                increase is.
            </Typography>
        </article>
    </MathJax.Provider>
)