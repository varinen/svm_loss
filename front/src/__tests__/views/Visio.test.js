import React from 'react';
import {shallow} from 'enzyme';
import Visio from '../../views/SVMLoss/Demo/Visio';

describe('Visio', () => {
    const component = shallow(<Visio/>);

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });
});
