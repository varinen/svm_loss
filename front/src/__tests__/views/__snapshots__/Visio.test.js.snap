// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visio renders correctly 1`] = `
<Card>
  <CardHeader
    color="info"
  >
    Data and Classifier Visualization
  </CardHeader>
  <CardBody
    className="makeStyles-pt2-3"
  >
    <_default />
    <Connect(WithStyles(Plot)) />
  </CardBody>
</Card>
`;
